### PR TITLE
style(lint): use Airbnb style guide

### PR DESCRIPTION
**Pull request by eventualbuddha**
This commit re-does our linting for all TypeScript code in the repo. It consolidates all configuration into `eslint-plugin-vx`. There are now two configuration presets: `recommended` and `react`. React apps (or libraries) should use `react`, and everything else should use `recommended`. Configuration within packages is now much simpler. Each `.eslintrc.js` has been replaced with a much slimmer `.eslintrc.json`, with any remaining boilerplate mostly related to those apps using Cypress.

The Airbnb style guide has two accompanying ESLint configs/plugins: `eslint-plugin-airbnb` and `eslint-plugin-airbnb-base`. The former includes React-related rules, so I only use that one in our `react` config. The latter is just the basic JavaScript rules and so forms the base of our `recommended` config. I tried to go with as vanilla as I could and to not customize our rules more than absolutely necessary beyond adding Prettier, which necessarily disables the Airbnb formatting and whitespace rules. However, Prettier and Airbnb are almost completely aligned on these things so it's not much of a change. The bigger consequences of not customizing the rules are things like `no-shadow`, which prohibit re-using a name from an outer scope. Turns out we did that a lot, and required a fair number of changes. Another was a rule that prohibits multiple classes per file, which required some refactoring.

Many of the changes in this commit are from disallowing TypeScript non-null assertions (postfix `!`). Instead, I added explicit `assert` calls to help TypeScript narrow the type and cause exceptions at the earliest point possible. In doing so I made some mistakes, like asserting on a value of type `number | undefined`, which would crash on falsy values that were nonetheless valid. So I added a new custom ESLint rule that checks that the value given to `assert` is not a `string` or `number`, requiring us to do a `typeof` check or some other suitable check. In a few places that made heavy use of non-null assertions (e.g.  tally code) I had to make some non-trivial refactors. These areas deserve special scrutiny in review.

Other minor things that required small refactors were `no-use-before-define`, `no-plusplus`, and `default-case`. Most of these were easy changes, but a few were non-trivial.

I also took the opportunity to relocate `iterators.ts` from `hmpb-interpreter` into `/utils`. One place in `module-scan` was using `zip` by reaching into the `dist` directory in `hmpb-interpreter`, and another area in `cardTallies.ts` seemed like it would be better done with `zip` so I refactored that.

Refs #764

### CODE REVIEW

File: apps/election-manager/src/components/SaveFileToUSB.tsx [hunk 1]
```diff
@@ -72,10 +70,11 @@ const SaveFileToUSB = ({
         fileDownload(results, defaultFilename, 'text/csv')
       } else {
         const usbPath = await usbstick.getDevicePath()
+        assert(typeof usbPath !== 'undefined')
```
Comment (carolinemodic) [inline apps/election-manager/src/components/SaveFileToUSB.tsx:73]: If you are in dev but running with kiosk browser its possible that usbPath is not defined here if you are using the manual save option when there is no usb. The way I wrote this conditional here doesn't make a lot of sense though, added suggestions for a cleaner way. If openFileDialog is true then we're in manual save mode and we don't need to worry about the usb path and if its false then usbPath must be defined.

Comment (carolinemodic) [reply apps/election-manager/src/components/SaveFileToUSB.tsx:73]:
```suggestion
```

File: apps/election-manager/src/components/SaveFileToUSB.tsx [hunk 2]
```diff
@@ -89,7 +88,8 @@ const SaveFileToUSB = ({
           setSavedFilename(fileWriter.filename)
           await fileWriter.end()
         } else {
-          await window.kiosk!.writeFile(pathToFile, results)
+          assert(window.kiosk)
```
Comment (carolinemodic) [inline apps/election-manager/src/components/SaveFileToUSB.tsx:91]:
```suggestion
          assert(typeof usbPath !== 'undefined')
          const pathToFile = path.join(usbPath, defaultFilename)
          assert(window.kiosk)
```

File: apps/election-manager/src/components/SaveFileToUSB.tsx [hunk 3]
```diff
@@ -72,10 +70,11 @@ const SaveFileToUSB = ({
...
         const usbPath = await usbstick.getDevicePath()
+        assert(typeof usbPath !== 'undefined')
         const pathToFile =
           !usbPath && process.env.NODE_ENV === 'development'
             ? defaultFilename
-            : path.join(usbPath!, defaultFilename)
+            : path.join(usbPath, defaultFilename)
         if (openFileDialog) {
           const fileWriter = await window.kiosk.saveAs({
             defaultPath: pathToFile,
```
Comment (carolinemodic) [inline apps/election-manager/src/components/SaveFileToUSB.tsx:80]:
```suggestion
            defaultPath: defaultFilename,
```

**Review by carolinemodic — APPROVED** [review]
Tested various election manager functions locally, just found one small issue in SaveFileToUSB.

File: apps/election-manager/src/lib/votecounting.test.ts [hunk 4]
```diff
@@ -40,15 +40,17 @@ export function parseCVRsAndAssertSuccess(
 function expectAllEmptyTallies(tally: Tally) {
   expect(tally.numberOfBallotsCounted).toBe(0)
   for (const contestId in tally.contestTallies) {
-    const contestTally = tally.contestTallies[contestId]!
-    for (const tally of Object.values(contestTally.tallies)) {
-      expect(tally!.tally).toBe(0)
+    if (Object.prototype.hasOwnProperty.call(tally.contestTallies, contestId)) {
+      const contestTally = tally.contestTallies[contestId]!
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/election-manager/src/lib/votecounting.test.ts:44]: Yeah, so they're actually still allowed in tests. I forgot that I did that 😄

**Pull request merged**